packages/jobs/utils/contract-utils.js

```diff
@@ -1,57 +1,79 @@
-const Web3 = require('web3');
-const Promise = require('bluebird');
 const contract = require('truffle-contract');
+const ethers = require('ethers');
 
 const config = require('../config');
 const logger = require('./logger')('utils/contract-utils');
 
-let ContractUtils = {};
-
 const contractsFolder = '../build/contracts/';
-let web3 = null;
+const MAX_ITERATIONS_FOR_WALLET_SEARCHING = 10000;
+const LOG_INTERATIONS_INTERVAL = 1000;
+
+let ContractUtils = {
+  getProvider,
+};
 
-const AliceToken = initializeContract('AliceToken');
-const ClaimsRegistry = initializeContract('ClaimsRegistry');
+ContractUtils.mainWallet = getMainWallet();
+validateNetworkId(ContractUtils.mainWallet.provider);
 
-ContractUtils.getClaimsRegistry = (address) => {
-  return getContractInstance(ClaimsRegistry, address);
+ContractUtils.getTruffleContract = function (name) {
+  const artefacts = require(contractsFolder + name + '.json');
+  return contract(artefacts);
 };
 
-ContractUtils.getProvider = function () {
-  lazyInitWeb3();
-  return web3.currentProvider;
+ContractUtils.getContractInstance = function (
+  contractName,
+  address,
+  wallet=ContractUtils.mainWallet
```

Comment:
I've noticed a pattern that you're manually creating a wallet(or rather signer) and then call getContractInstance. 
Could we keep the logic in here and pass an address as an argument for simplicity?

Reply:
Now I see "The Wallet implements the Signer API". 
I still think that it will be more readable if we use wallet for a collection of addresses and signer for a single-address generated wallet. What do you think?